internal/utils/config.go

```diff
@@ -0,0 +1,92 @@
+package utils
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+// Configuration holds the configuration to connect to the Argo CD server
+type Configuration struct {
+	ArgoCDAuthToken  string
+	ArgoCDServerAddr string
+	ArgoCDInsecure   bool
+}
+
+// readFromFile returns the content of a file
+func readFromFile(path string) (string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+
+	data, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+
+	return string(data), nil
+}
+
+// readFromEnvOrFile returns the value of an environment variable if present, or from a file otherwise
+func readFromEnvOrFile(paramName string) (string, error) {
+	var err error
+	value := os.Getenv(paramName)
+
+	if len(value) == 0 {
+		value, err = readFromFile(ConfigDirectory + paramName)
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
```suggestion
	value, ok := os.LookupEnv(paramName)

	if !ok {
		value, err = readFromFile(ConfigDirectory + paramName)
		if err != nil {
			return "", err
		}
	}
```
If you use `LookupEnv` you get also a nice bool to tell you if the environment variable is set or not. What do you think?